torch/nn/parallel/distributed.py

```diff
@@ -925,8 +925,15 @@ def forward(self, *inputs, **kwargs):
                 self._check_global_requires_backward_grad_sync(is_joined_rank=False)
 
             if self.device_ids:
+                isLazy = inputs[0].device.type == 'lazy'
```

Comment:
i'm confused what this part is for. 

1) the model/weights/grads from demo script should already be lazy tensors right?  - so is this part trying to move tensors to the correct device index for some reason?
2) aren't you just going to convert back to cuda inside c++?

Reply:
`self.to_kwargs` moves things back to cuda devices... Annoying...